### PR TITLE
Don't generate plugin registry in ResidentWebRunner

### LINKED ISSUE #91262: Regression in restart time benchmarks
SkiaPerf has identified a regression in restart time metrics:

https://flutter-flutter-perf.skia.org/e/?begin=1632536886&end=1633382183&keys=X9da79a237b5af9f0d5f350d84cee3803&requestType=0&xbaroffset=26039

The implicated change is https://github.com/flutter/flutter/pull/87991. Please consider reverting since we are trying to cut the beta branch this week.

/cc @blasten

### PR DESCRIPTION
**Pull request by stuartmorgan-g**
generateDartPluginRegistry was being set to true unconditionally in
ResidentRunner, bypassing the primary check in
DartPluginRegistrantTarget, and the targetPlatform was not set in that
codepath, bypassing the second after the changes in
https://github.com/flutter/flutter/pull/87991. This caused web hot
restarts to be slower due to doing unnecessary work.

This ensures that generateDartPluginRegistry is false in the
ResidentWebRunner to skip that unnecessary step.

Fixes https://github.com/flutter/flutter/issues/91262

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by zanderso** [comment]
> /cc @zanderso @christopherfujino FYI. I'm not sure how to run the perf test myself, but I verified locally that the undesired codepath was being run for web hot restart without this change, as is not with this change.

I don't know of an easier way of seeing how to run the benchmarks than looking at how the bots run them. In this case you can search on the dashboard for "chrome_dev_mode", which leads you to and example run:

https://ci.chromium.org/ui/p/flutter/builders/prod/Mac_ios%20macos_chrome_dev_mode/2408/overview

From there, you probably just need to look at step 17 "run macos_chrome_dev_mode".

File: packages/flutter_tools/test/general.shard/resident_web_runner_test.dart [hunk 1]
```diff
@@ -941,6 +941,44 @@ void main() {
...
+      "name": "path_provider_linux",
+      "rootUri": "../../../path_provider_linux",
+      "packageUri": "lib/",
+      "languageVersion": "2.12"
+    }
+  ]
+}
+''');
+    // Start with a generated_main.dart file.
+    globals.fs.directory('.dart_tool').childDirectory('flutter_build').childFile('generated_main.dart').createSync(recursive: true);
```
Comment (zanderso) [inline packages/flutter_tools/test/general.shard/resident_web_runner_test.dart:965]:
```suggestion
    globals.fs.directory('.dart_tool')
              .childDirectory('flutter_build')
              .childFile('generated_main.dart')
              .createSync(recursive: true);
```

**Review by zanderso — APPROVED** [review]
lgtm w/ nit

**Comment by zanderso** [comment]
@godofredoc the ci.yaml validation looks like it is taking too long to queue on this PR.

**Comment by godofredoc** [comment]
> @godofredoc the ci.yaml validation looks like it is taking too long to queue on this PR.
This is a problem with the request to trigger the builds timing out. Retried all builds.

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 2.
View them at https://flutter-gold.skia.org/cl/github/91281

**Review by blasten — APPROVED** [review]

**Comment by zanderso** [comment]
Any idea why this change affects goldens?

**Comment by blasten** [comment]
That is weird. maybe a rebase fixes it?

**Comment by stuartmorgan** [comment]
> That is weird. maybe a rebase fixes it?

Yep! No idea what happened there; infra issue maybe?

**Comment by blasten** [comment]
I think Skia Gold's state is not a function of the files in your local copy. It's a function of whomever approved the last gold files?

**Pull request merged**